pkgs/development/python-modules/pybind11/default.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+  pname = "pybind11";
+  version = "2.2.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1kz1z2cg3q901q9spkdhksmcfiskaghzmbb9ivr5mva856yvnak4";
+  };
+
+  meta = {
+    homepage = https://github.com/pybind/pybind11;
+    description = "Seamless operability between C++11 and Python";
+    longDescription = ''
+      Pybind11 is a lightweight header-only library that exposes
+      C++ types in Python and vice versa, mainly to create Python
+      bindings of existing C++ code.
```

Comment:
Since this package - if I understand it correctly - can only be used in one direction, this shouldn't mention both.

Reply:
I check the official documentation again and the python extension can also be used in the second use case through python instead of CMake.

As the package still includes the header, the simplest use case would look something like this:
```bash
c++ -O3 -Wall -shared -std=c++11 -fPIC `python3 -m pybind11 --includes` example.cpp -o example`python3-config --extension-suffix`
```
As a result pybind11 can easily be used with ```setuptools``` (see [example](https://github.com/pybind/python_example)) with full PyPi support.